patches/expo-notifications+0.28.1.patch

```diff
@@ -0,0 +1,325 @@
+ 
+ import expo.modules.notifications.notifications.model.Notification;
+@@ -28,6 +30,14 @@ public interface NotificationListener {
+     return false;
+   }
+ 
++  /**
++   * Callback called when notification response is received through package lifecycle listeners
++   *
++   * @param extras Bundle of extras from the lifecycle method
++   */
++  default void onNotificationResponseIntentReceived(Bundle extras) {
++  }
++
+   /**
+    * Callback called when some notifications are dropped.
+    * See {@link FirebaseMessagingService#onDeletedMessages()}
+diff --git a/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java
+new file mode 100644
+index 0000000..55e3ab9
+--- /dev/null
++++ b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/ExpoNotificationLifecycleListener.java
+@@ -0,0 +1,79 @@
++package expo.modules.notifications.service.delegates;
++
++import android.app.Activity;
++import android.app.NotificationChannel;
++import android.app.PendingIntent;
++import android.content.Context;
++import android.content.Intent;
++import android.os.Bundle;
++import android.os.Parcel;
++import android.util.Log;
++
++import androidx.core.app.NotificationCompat;
++
++import expo.modules.core.interfaces.ReactActivityLifecycleListener;
++import expo.modules.notifications.notifications.NotificationManager;
++import expo.modules.notifications.notifications.model.Notification;
++import expo.modules.notifications.notifications.model.NotificationResponse;
++
++public class ExpoNotificationLifecycleListener implements ReactActivityLifecycleListener {
++
++    private Context mContext;
++    private NotificationManager mNotificationManager;
++
++    public ExpoNotificationLifecycleListener(Context context, NotificationManager notificationManager) {
++      mContext = context;
++      mNotificationManager = notificationManager;
++    }
++
++    /**
++     * This will be triggered if the app is not running,
++     * and is started from clicking on a notification.
++     * <p>
++     * Notification data will be in activity.intent.extras
++     *
++     * @param activity
++     * @param savedInstanceState
++     */
++    @Override
++    public void onCreate(Activity activity, Bundle savedInstanceState) {
++        Intent intent = activity.getIntent();
++        String actionIdentifier = intent.getAction();
++        if (intent != null) {
++            Bundle extras = intent.getExtras();
++            if (extras != null) {
++                logExtra("onCreate", extras);
++                mNotificationManager.onNotificationResponseFromExtras(extras);
++            }
++        }
++        ReactActivityLifecycleListener.super.onCreate(activity, savedInstanceState);
```

Comment:
We don't need to call `super` here, it just calls an empty implementation. These listeners are called from inside the lifecycle methods in `ReactActivityDelegateWrapper.kt`.

Reply:
Fixed